cassandra/cluster.py

```diff
@@ -165,10 +165,12 @@ def _connection_reduce_fn(val,import_fn):
         excs.append(exc)
     return (rv or import_result, excs)
 
+log = logging.getLogger(__name__)
+
 conn_fns = (_try_gevent_import, _try_eventlet_import, _try_libev_import, _try_asyncore_import)
 (conn_class, excs) = reduce(_connection_reduce_fn, conn_fns, (None,[]))
-if excs:
-    raise DependencyException("Exception loading connection class dependencies", excs)
+if not conn_class:
+    raise DependencyException("Unable to load a default connection class", excs)
```

Comment:
A fix for an issue discovered in testing.  Goal of the changes to support asyncore as an optional dependency was to raise an error if we were unable to find a workable event loop at all.  Unfortunately my original impl just looked for the presence of any exception in any event loop impl which is... not at all the same thing.